Improvement/cldsrv 553 bump deps

benzekrimaha opened the pull request:
Issue: CLDSRV-553

bert-e commented:
# Hello benzekrimaha,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://bert-e.scality.net/scality/cloudserver/doc/user).


<details>
  <summary><b>Available options</b></summary>

  name   | description  | privileged   | authored
  ------ | ------------ | ------------ |----------
  `/after_pull_request` | Wait for the given pull request id to be merged before continuing with the current one. |  | 
  `/bypass_author_approval` | Bypass the pull request author's approval |  :star:  | 
  `/bypass_build_status` | Bypass the build and test status |  :star:  | 
  `/bypass_commit_size` | Bypass the check on the size of the changeset ```TBA``` |  :star:  | 
  `/bypass_incompatible_branch` | Bypass the check on the source branch prefix |  :star:  | 
  `/bypass_jira_check` | Bypass the Jira issue check |  :star:  | 
  `/bypass_peer_approval` | Bypass the pull request peers' approval |  :star:  | 
  `/bypass_leader_approval` | Bypass the pull request leaders' approval |  :star:  | 
  `/approve` | Instruct Bert-E that the author has approved the pull request. |  |  :writing_hand: 
  `/create_pull_requests` | Allow the creation of integration pull requests. |  | 
  `/create_integration_branches` | Allow the creation of integration branches. |  | 
  `/no_octopus` | Prevent Wall-E from doing any octopus merge and use multiple consecutive merge instead |  | 
  `/unanimity` | Change review acceptance criteria from `one reviewer at least` to `all reviewers` |  | 
  `/wait` | Instruct Bert-E not to run until further notice. |  | 
  

</details>



<details>
  <summary><b>Available commands</b></summary>

  name   | description  | privileged
  ------ | ------------ | ------------
  `/help` | Print Bert-E's manual in the pull request. | 
  `/status` | Print Bert-E's current status in the pull request ```TBA``` | 
  `/clear` | Remove all comments from Bert-E from the history ```TBA``` | 
  `/retry` | Re-start a fresh build ```TBA``` | 
  `/build` | Re-start a fresh build ```TBA``` | 
  `/force_reset` | Delete integration branches & pull requests, and restart merge process from the beginning. | 
  `/reset` | Try to remove integration branches unless there are commits on them which do not appear on the source branch. | 
  
</details>




*Status report is not available.*

bert-e commented:
# Incorrect fix version



The `Fix Version/s` in issue CLDSRV-553 contains:


* *None*



Considering where you are trying to merge, I ignored possible hotfix versions and I expected to find:



* `8.8.39`


Please check the `Fix Version/s` of CLDSRV-553, or the target
branch of this pull request.

bert-e commented:
# Incorrect fix version



The `Fix Version/s` in issue CLDSRV-553 contains:


* *None*



Considering where you are trying to merge, I ignored possible hotfix versions and I expected to find:



* `8.9.0`


Please check the `Fix Version/s` of CLDSRV-553, or the target
branch of this pull request.

benzekrimaha committed:
Bump dependencies

Issue: CLDSRV-553

benzekrimaha committed:
Adapt docker file

Issue: CLDSRV-553

benzekrimaha committed:
Adapt config post bump eslint to v9

Issue: CLDSRV-553

francoisferrand commented on Dockerfile:
```diff
@@ -1,4 +1,4 @@
-ARG NODE_VERSION=16.20.2-bookworm-slim
+ARG NODE_VERSION=22-bookworm-slim
```
should stick to a specific node version, to ensure reproductible & repeatable builds

bert-e commented:
# Conflict




There is a conflict between your branch `improvement/CLDSRV-553-bump-deps` and the
destination branch `development/8.9`.

Please resolve the conflict on **the feature branch** (`improvement/CLDSRV-553-bump-deps`).

```sh
git fetch && \
git checkout origin/improvement/CLDSRV-553-bump-deps && \
git merge origin/development/8.9
```

Resolve merge conflicts and commit

```sh
git push origin HEAD:improvement/CLDSRV-553-bump-deps
```

benzekrimaha committed:
Update code to match scality guidelines

Issue: CLDSRV-553

benzekrimaha committed:
Md lint fixup

Issue: CLDSRV-553

benzekrimaha committed:
Adapt package.json mocha tests commands

Issue: CLDSRV-553

francoisferrand commented on Dockerfile:
```diff
@@ -22,6 +22,7 @@ RUN apt-get update \
     && ssh-keyscan -H github.com > /root/ssh/known_hosts
 
 ENV PYTHON=python3
+RUN npm install -g node-gyp
 COPY package.json yarn.lock /usr/src/app/
 RUN npm install typescript@4.9.5 -g
```
best to install all "global" packages in a single step,
before package.json copy (to avoid invalidatoin),
and specifying the node-gyp version (just like we do for typescript)

```suggestion
RUN npm install -g \
    node-gyp@11.0.0 \
    typescript@4.9.5
COPY package.json yarn.lock /usr/src/app/
```

francoisferrand replied:
ping

benzekrimaha committed:
Tests fixups post bump failures

Issue: CLDSRV-553

benzekrimaha committed:
Bump node version in gha

Issue: CLDSRV-553

francoisferrand commented on lib/api/apiUtils/authorization/prepareRequestContexts.js:
```diff
@@ -154,13 +154,6 @@ function prepareRequestContexts(apiMethod, request, sourceBucket,
                 generateRequestContext('objectPutTaggingVersion');
             requestContexts.push(putObjectVersionRequestContext);
         }
```
why are we changing permissions?
if it was (already) a bug, it shoudl be fixed on 8.8 as well instead here...

francoisferrand replied:
@benzekrimaha did you miss this one?

benzekrimaha replied:
it's not changing the permission ,this code is actually duplicated see here : https://github.com/scality/cloudserver/pull/5711/files#diff-465d8f1849b67c1a40847391c0a7ea074b6357a8b7cdeb2db2be89ad9dfce50bL79-L80

we are never going to cross this condition

francoisferrand commented on lib/api/apiUtils/authorization/tagConditionKeys.js:
```diff
@@ -61,7 +61,6 @@ function updateRequestContextsWithTags(request, requestContexts, apiMethod, log,
         }
         // FIXME introduced by CLDSRV-256, this syntax should be allowed by the linter
-        // eslint-disable-next-line no-restricted-syntax
         for (const rc of requestContexts) {
```
instead of disabling the linter, we should either update the linter configuration (to allow this), or retry this in a conforming maner : `requestContext.each(rc -> { ... })`

benzekrimaha committed:
Skipping tests not working

Issue: CLDSRV-553

francoisferrand commented on lib/Config.js:
```diff
@@ -1220,145 +1175,6 @@ class Config extends EventEmitter {
             maxStaleness,
             enableInflights,
         };
-        if (config.utapi) {
```
we should keep the code, only need to replace the condition to make sure it never happens:


```suggestion
        if (false && config.utapi) {
```

(and we can keep the `parseUtapiReindex` function as well)

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* 2 peers

francoisferrand commented on lib/utapi/utilities.js:
i think it may be better to keep Utapi in the code:
* only need to comment these few lines  the `require(utapi)` add an early retrun in `pushMetrics`
* the change will be much smaller (no need to change all places where this module is used), and thus easier to review & identify node22-related changes
* any change to utapi-code (in 7.x) will still be propagated with waterflow
* we can fix lint issues already, saving time when actually bumping utapi
* having it in a commit still makes it easy to revert, once utapi is upgraded

francoisferrand commented on .github/workflows/tests.yaml:
```diff
@@ -342,39 +342,6 @@ jobs:
         source: /tmp/artifacts
       if: always()
 
```
should be skipped instead of removed:

```suggestion
  utapi-v2-tests:
    if: false
```

benzekrimaha committed:
Revert "[to be reverted]utapi removal"

This reverts commit a20bc5d8dd7ed4d3bb0de972ae8f4fdacf26f0ed.

benzekrimaha committed:
test

benzekrimaha committed:
fixup

francoisferrand commented:
please rebase, there are conflicts.
(sorry, add some pending commensts, which may not be relevant anymore : please with this in mind...)

francoisferrand commented on bin/list_metrics.js:
```diff
@@ -1,4 +0,0 @@
-#!/usr/bin/env node
```
should keep this file, and just add early return (if needed) in `listMetrics()`

francoisferrand commented on bin/list_bucket_metrics.js:
```diff
@@ -1,4 +0,0 @@
-#!/usr/bin/env node
-'use strict'; // eslint-disable-line strict
-
-require('../lib/utapi/utilities.js').listMetrics('buckets');
```
should keep this file, and just add early return (if needed) in `listMetrics()`

benzekrimaha committed:
pot review docker file fixup

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* 2 peers

benzekrimaha committed:
fixup

benzekrimaha committed:
fixup

benzekrimaha committed:
fixup

benzekrimaha committed:
fixup

francoisferrand commented on .github/workflows/tests.yaml:
```diff
@@ -361,38 +361,38 @@ jobs:
-        password: ${{ secrets.ARTIFACTS_PASSWORD }}
-        source: /tmp/artifacts
-      if: always()
+  # utapi-v2-tests:
```
better to skip the job than to comment it:

```suggestion
  utapi-v2-tests:
    if: false
    [...]

bert-e commented:
# Conflict




There is a conflict between your branch `improvement/CLDSRV-553-bump-deps` and the
destination branch `development/9.0`.

Please resolve the conflict on **the feature branch** (`improvement/CLDSRV-553-bump-deps`).

```sh
git fetch && \
git checkout origin/improvement/CLDSRV-553-bump-deps && \
git merge origin/development/9.0
```

Resolve merge conflicts and commit

```sh
git push origin HEAD:improvement/CLDSRV-553-bump-deps
```

francoisferrand commented on .github/workflows/tests.yaml:
```diff
@@ -149,7 +149,7 @@ jobs:
 
   build:
-    runs-on: ubuntu-20.04
+    runs-on: ubuntu-latest
```
best to specify the version explicitely (24.04)

francoisferrand commented on lib/api/bucketPutACL.js:
```diff
@@ -261,6 +263,9 @@ function bucketPutACL(authInfo, request, log, callback) {
                         return next(null, bucket, revisedAddACLParams);
                     });
             }
+            if (hasError) {
```
should be earlier in the code (eright after the `jsonGrants.forEach` loop) : otherwise we may continue while we had an error, and risk calling the callback again (like on line 263)

francoisferrand commented on lib/api/objectPutACL.js:
```diff
@@ -272,6 +274,9 @@ function objectPutACL(authInfo, request, log, cb) {
                             revisedAddACLParams);
                     });
             }
+            if (hasError) {
```
same, should be called earlier : to avoid having a duplicate call via line 273...

francoisferrand commented on lib/utapi/utapiReindex.js:
```diff
@@ -1,11 +1,11 @@
-    reindexConfig.password = config.utapi && config.utapi.redis
-    && config.utapi.redis.password;
-}
-const reindex = new UtapiReindex(reindexConfig);
-reindex.start();
+// const reindexConfig = config.utapi && config.utapi.reindex;
+// if (reindexConfig && reindexConfig.password === undefined) {
+//     reindexConfig.password = config.utapi && config.utapi.redis
+//     && config.utapi.redis.password;
+// }
```
no need to comment these

francoisferrand commented on lib/utapi/utapiReplay.js:
```diff
@@ -1,8 +1,8 @@
 
-const utapiConfig = _config.utapi &&
-    Object.assign({}, _config.utapi, { redis: _config.redis });
-const replay = new UtapiReplay(utapiConfig); // start utapi server
-replay.start();
+// const utapiConfig = _config.utapi &&
+//     Object.assign({}, _config.utapi, { redis: _config.redis });
```
no need to comment

francoisferrand commented on lib/Config.js:
```diff
@@ -13,14 +13,11 @@ const { isValidBucketName } = s3routes.routesUtils;
-    allowedUtapiEventFilterFields,
-    allowedUtapiEventFilterStates,
-    supportedLifecycleRules,
+const { azureAccountNameRegex, base64Regex,
+    allowedUtapiEventFilterFields, allowedUtapiEventFilterStates,
+    supportedLifecycleRules
 } = require('../constants');
```
change not needed?

francoisferrand replied:
?

francoisferrand commented on lib/Config.js:
```diff
@@ -1236,7 +1233,7 @@ class Config extends EventEmitter {
             enableInflights,
         };
-        if (config.utapi) {
+        if (config.utapi && false) { // eslint-disable-line no-constant-condition
```
no need to change, will work in case we don't have utapi?

francoisferrand commented on lib/Config.js:
```diff
@@ -1368,8 +1365,9 @@ class Config extends EventEmitter {
+        // eslint-disable-next-line no-constant-condition
         if (Object.keys(this.locationConstraints).some(
-        loc => this.locationConstraints[loc].sizeLimitGB)) {
+        loc => this.locationConstraints[loc].sizeLimitGB) && false) {
```
no need to 'fase' this, we should never _run_ into this code anyway

francoisferrand commented on lib/Config.js:
```diff
@@ -3,7 +3,7 @@ const { EventEmitter } = require('events');
 const path = require('path');
 const url = require('url');
-const crypto = require('crypto');
+const mycrypto = require('crypto');
```
why rename?
can't we use node22's new built-in crypto module?

francoisferrand commented on tests/functional/aws-node-sdk/lib/utility/createEncryptedBucket.js:
```diff
@@ -1,5 +1,5 @@
 const childProcess = require('child_process');
-const Promise = require('bluebird');
+const myPromise = require('bluebird');
```
rename `bluebirdPromise` (or `bbPromise`).... or can we switch to using "standard" node.js Promise instead ?

francoisferrand commented on tests/functional/aws-node-sdk/test/bucket/get.js:
```diff
@@ -1,6 +1,6 @@
 const assert = require('assert');
 const tv4 = require('tv4');
-const Promise = require('bluebird');
+const myPromise = require('bluebird');
```
same

francoisferrand commented on tests/functional/aws-node-sdk/test/legacy/authV2QueryTests.js:
```diff
@@ -1,5 +1,5 @@
 const assert = require('assert');
-const process = require('process');
+const myProcess = require('process');
```
can't we switch to using the new standard process module?

```
const process = require('node:process');
```

francoisferrand commented on tests/functional/aws-node-sdk/test/legacy/authV4QueryTests.js:
```diff
@@ -1,5 +1,5 @@
 const assert = require('assert');
-const process = require('process');
+const myProcess = require('process');
```
same

francoisferrand commented on tests/functional/aws-node-sdk/test/multipleBackend/objectCopy/azureObjectCopy.js:
```diff
@@ -31,7 +31,7 @@ const bigMD5 = '5f363e0e58a95f06cbe9bbc662c5dfb6';
 const locMetaHeader = constants.objectLocationConstraintHeader.substring(11);
 
-const Promise = require('bluebird');
+const myPromise = require('bluebird');
```
use default Promise & promisify?

francoisferrand commented on tests/functional/aws-node-sdk/test/multipleBackend/objectCopy/objectCopy.js:
```diff
@@ -19,7 +19,7 @@ const body = Buffer.from('I am a body', 'utf8');
 const emptyMD5 = 'd41d8cd98f00b204e9800998ecf8427e';
 const locMetaHeader = constants.objectLocationConstraintHeader.substring(11);
-const Promise = require('bluebird');
+const myPromise = require('bluebird');
```
use default Promise & promisify?

francoisferrand commented on tests/functional/aws-node-sdk/test/legacy/tests.js:
```diff
@@ -1,5 +1,5 @@
 const assert = require('assert');
-const crypto = require('crypto');
+const myCrypto = require('crypto');
```
use default node:crypto ?

francoisferrand commented on tests/functional/aws-node-sdk/test/multipleBackend/put/put.js:
```diff
@@ -11,7 +11,7 @@ const { versioningEnabled } = require('../../../lib/utility/versioning-util');
     awsLocationEncryption, memLocation, fileLocation, genUniqID, isCEPH }
     = require('../utils');
-const Promise = require('bluebird');
+const myPromise = require('bluebird');
```
same

francoisferrand commented on tests/functional/aws-node-sdk/test/multipleBackend/utils.js:
```diff
@@ -1,5 +1,5 @@
 const assert = require('assert');
-const crypto = require('crypto');
+const myCrypto = require('crypto');
```
same

francoisferrand commented on tests/functional/aws-node-sdk/test/object/copyPart.js:
```diff
@@ -1,7 +1,7 @@
 const assert = require('assert');
-const crypto = require('crypto');
+const myCrypto = require('crypto');
 
-const Promise = require('bluebird');
+const myPromise = require('bluebird');
 
```
same

francoisferrand commented on tests/functional/aws-node-sdk/test/quota/tooling.js:
```diff
@@ -33,18 +33,20 @@ const sendRequest = async (method, host, path, body = '', config = null) =>
-                reject(result);
-            } else {
-                resolve(result);
+        (async () => {
```
why do we need this?

since the code is already async/wait, we can simply remove the outer Promise, and keep the function async:

```
const sendRequest = async (method, host, path, body = '', config = null) =>
    const service = 's3';
    const endpoint = new AWS.Endpoint(host);
   
    [...]
   
    const response = await nodeFetch(url, options);
    const text = await response.text();
    const result = await xml2js.parseStringPromise(text);
    if (result && result.Error) {
        throw result;
    }

    return result;
}
```

francoisferrand commented on tests/functional/metadata/MixedVersionFormat.js:
```diff
@@ -97,7 +97,9 @@ describe('Mongo backend mixed bucket format versions', () => {
         });
 
-        after(done => mongoClient.close(true).then(() => done()).catch(() => done()));
+        after(() => {
+            mongoClient.close(true);
+        });
 
```
mongodClient.close() is async (ie returns a Promise), so need to make the function passed to after async as well: too ensure it waits until mongo client is closed.

```suggestion
        after(() => mongoClient.close(true));
```

or

```suggestion
        after(async () => {
            await mongoClient.close(true);
        });
```

francoisferrand commented on tests/functional/metadata/MongoClientInterface.js:
```diff
@@ -44,7 +44,7 @@ const updatedObjVal = { updated: true };
     process.env.S3METADATA === 'mongodb' ? describe : describe.skip;
 
-function unescape(obj) {
+function unescapeFunc(obj) {
```
```suggestion
function unescapeJSON(obj) {
```

francoisferrand commented on tests/functional/raw-node/utils/HttpRequestAuthV4.js:
```diff
@@ -1,11 +1,11 @@
-const crypto = require('crypto');
+const myCrypto = require('crypto');
```
use node:crypto?

francoisferrand commented on tests/functional/s3cmd/tests.js:
```diff
@@ -1,7 +1,7 @@
 
 const proc = require('child_process');
-const process = require('process');
+const myProcess = require('process');
```
use node:process ?

francoisferrand commented on tests/functional/s3curl/tests.js:
```diff
@@ -1,8 +1,8 @@
 const assert = require('assert');
 const proc = require('child_process');
-const process = require('process');
+const myProcess = require('process');
 const parseString = require('xml2js').parseString;
```
same

francoisferrand commented on tests/multipleBackend/routes/routeBackbeat.js:
```diff
@@ -1,7 +1,7 @@
 const AWS = require('aws-sdk');
 const async = require('async');
-const crypto = require('crypto');
+const myCrypto = require('crypto');
```
use node:crypto?

francoisferrand commented on tests/multipleBackend/routes/routeVeeam.js:
```diff
@@ -1,5 +1,5 @@
 const assert = require('assert');
-const crypto = require('crypto');
+const myCrypto = require('crypto');
```
same

francoisferrand commented on tests/quota/awsNodeSdk.js:
```diff
@@ -338,11 +338,18 @@ function multiObjectDelete(bucket, keys, size, callback) {
 
-        before(done => {
+        before(async () => {
             const config = getConfig('default', { signatureVersion: 'v4', maxRetries: 0 });
             s3Client = new S3(config);
             scuba.start();
-            return metadata.setup(err => wait(2000, () => done(err)));
+            await new Promise((resolve, reject) => {
+                metadata.setup(err => {
+                    if (err) {
+                        reject(err);
+                    }
+                    wait(2000, resolve);
+                });
+            });
         });
```
```suggestion
        before(() => {
            const config = getConfig('default', { signatureVersion: 'v4', maxRetries: 0 });
            s3Client = new S3(config);
            scuba.start();
            return new Promise((resolve, reject) => {
                metadata.setup(err => {
                    if (err) {
                        reject(err);
                    }
                    wait(2000, resolve);
                });
            });
        });
```

francoisferrand commented on tests/unit/api/objectReplicationMD.js:
```diff
@@ -77,7 +77,17 @@ const emptyReplicationMD = {
-    isNFS: null,
+    isNFS: undefined,
+};
+const emptyReplicationMDUndef = {
```
did you mean `emptyReplicationInfo` ?

francoisferrand commented on tests/unit/api/objectReplicationMD.js:
```diff
@@ -77,7 +77,17 @@ const emptyReplicationMD = {
-    isNFS: null,
+    isNFS: undefined,
+};
+const emptyReplicationMDUndef = {
```
I don't think we put replicationInfo with `isNFS: undefined` : same (empty) fields are indeed not returned anymore (change in arsenal), but we did not change the way we write the metadata, so the "input" parameter of the tests should not change


```suggestion
    isNFS: null,
};
const expectedEmptyReplicationMD = {
```

benzekrimaha replied:
The test is failing otherwise waiting for undefined instead of nul

francoisferrand replied:
> we did not change the way we write the metadata, so the "input" parameter of the tests should not change

if the test fail, it is probably because we at some point compare to `emptyReplicationMD` instead of `emptyReplicationMDUndef` (which would be better renamed `expectedEmptyReplicationMD` I think)

francoisferrand commented on tests/unit/Config.js:
```diff
@@ -470,7 +470,7 @@ describe('Config', () => {
         });
 
-        it('should set up utapi local cache', () => {
+        it.skip('should set up utapi local cache', () => {
```
this should make linter fail : as it does not, it seems something is missing from the config :-(
so we'll need to think about adding it later...

benzekrimaha committed:
deps update

benzekrimaha committed:
post reviews fixups

benzekrimaha committed:
post reviews fixups

francoisferrand commented on lib/Config.js:
```diff
@@ -13,14 +13,11 @@ const { isValidBucketName } = s3routes.routesUtils;
+    supportedLifecycleRules
 } = require('../constants');
-const { utapiVersion } = require('utapi');
+// const { utapiVersion } = require('utapi');
```
"mock" the variable definition, so you don't need to add `eslint-disable-line no-undef`

```suggestion
const utapiVersion = 0; // require('utapi');
```

francoisferrand commented on tests/functional/aws-node-sdk/test/multipleBackend/objectCopy/azureObjectCopy.js:
```diff
@@ -31,7 +31,7 @@ const bigMD5 = '5f363e0e58a95f06cbe9bbc662c5dfb6';
 
-const Promise = require('bluebird');
+const bbPromise = require('bluebird');
 
```
maybe not for this PR, but from your experience here, do you think we could/should remove this dependency? would it be resonnable?

if so, please create a followup ticket

francoisferrand replied:
this is recomended by bluebird: https://github.com/petkaantonov/bluebird/#%EF%B8%8Fnote%EF%B8%8F

> Please use native promises instead if at all possible. Native Promises have been stable in Node.js and browsers for around 10 years now and they have been fast for around 7. Any utility bluebird has like .map has native equivalents (like Node streams' .map).
>
> This is a good thing, the people working on Bluebird and promises have been able to help incorporate most of the useful things from Bluebird into JavaScript itself and platforms/engines.
>
> If there is a feature that keeps you using bluebird. Please let us know so we can try and upstream it :)
>
> Currently - it is only recommended to use Bluebird if you need to support really old browsers or EoL Node.js or as an intermediate step to use warnings/monitoring to find bugs.

benzekrimaha replied:
indeed this needs to be reworked later on, here's the ticket created : https://scality.atlassian.net/browse/CLDSRV-605

francoisferrand commented on tests/functional/aws-node-sdk/test/quota/tooling.js:
```diff
@@ -1,9 +1,9 @@
-    new Promise(async (resolve, reject) => {
+const sendRequest = async (method, host, path, body = '', config = null) => {
+
         const service = 's3';
```
* extra empty line
* bad indent: should be just 4 spaces

francoisferrand commented on tests/quota/awsNodeSdk.js:
```diff
@@ -338,11 +338,18 @@ function multiObjectDelete(bucket, keys, size, callback) {
+            return new Promise((resolve, reject) => {
+                metadata.setup(err => {
+                    if (err) {
+                        reject(err);
```
missing `return`

francoisferrand commented on tests/quota/awsNodeSdk.js:
```diff
@@ -338,11 +338,18 @@ function multiObjectDelete(bucket, keys, size, callback) {
             scuba.start();
-            return metadata.setup(err => wait(2000, () => done(err)));
+            return new Promise((resolve, reject) => {
+                metadata.setup(err => {
+                    if (err) {
+                        reject(err);
+                    }
+                    wait(2000, resolve);
+                });
+            });
```
do we really _need_ to change the callback-based hook here?
as we are still using only callback-based functions internally, the updated code is more complex than the old one: if no constraint, may be better to keep it until those functions have been updated

francoisferrand reviewed (commented):
LGTM overall, but a few comments have not been adressed yet (esp. in gha)

benzekrimaha committed:
fixups post reviews

francoisferrand commented on tests/multipleBackend/routes/routeBackbeat.js:
```diff
@@ -292,16 +291,14 @@ describe('backbeat routes', () => {
             bucketUtil.emptyIfExists(BUCKET_FOR_NULL_VERSION)
                 .then(() => s3.createBucket({ Bucket: BUCKET_FOR_NULL_VERSION }).promise())
-                .then(() => done(), err => done(err))
         );
```
i don't understand how this code "waits" for the bucket to be created :(
(it would be the case if promise was returned, but i don't see how it is here...)

benzekrimaha commented:
/approve

benzekrimaha commented:
/wait

francoisferrand committed:
Rename eslint-config-scality to @scality org

This avoids the malware advisory published by Github [1].

[1] https://github.com/advisories/GHSA-52g9-9f8x-g5m5

Issue: CLDSRV-553

benzekrimaha commented:
closing pull request as https://github.com/scality/cloudserver/pull/5734 was merged
